src/realm/query_engine.hpp

```diff
@@ -1545,9 +1544,14 @@ class StringNodeBase : public ParentNode {
 template <class TConditionFunction>
 class StringNode : public StringNodeBase {
 public:
+    constexpr static bool case_sensitive_comparison =
```

Comment:
it is a compile error to remove the static on a constexpr member

Reply:
I was more thinking whether you needed it as `public` member or you could just have a `constexpr` variable defined in the method.

Reply:
Oh I see. Well I suppose it doesn't need to be public, but I'd like it at the class scope because I use it for a small optimization in the constructor as well.